Tasks/DeployVisualStudioTestAgent/DeployTestAgent.ps1

```diff
@@ -1,111 +1,117 @@
-
-Function CmdletHasMember($memberName) {
-    $cmdletParameter = (gcm Register-Environment).Parameters.Keys.Contains($memberName)
-    return $cmdletParameter
-}
-
-Function Get-PersonalAccessToken($vssEndPoint) {
-    return $vssEndpoint.Authorization.Parameters.AccessToken
-}
-
-Write-Verbose "Entering script DeployTestAgent.ps1"
-Write-Verbose "testMachineInput = $testMachineGroup"
-Write-Verbose "WinRmProtocal = $winRmProtocol"
-Write-Verbose "resourceFilteringMethod = $resourceFilteringMethod" -Verbose
-Write-Verbose "filter testMachines = $testMachines"
-Write-Verbose "runAsProcess = $runAsProcess"
-Write-Verbose "logonAutomatically = $logonAutomatically"
-Write-Verbose "disableScreenSaver = $disableScreenSaver"
-Write-Verbose "updateTestAgent = $updateTestAgent"
-Write-Verbose "isDataCollectionOnly = $isDataCollectionOnly"
-
-if ([string]::IsNullOrWhiteSpace($agentLocation))
-{
-   Write-Verbose "Download of testagent would begin from internet"
-}
-else
-{
-   Write-Verbose "agentLocation = $agentLocation"
-}
-
-# Get current directory.
-$currentDirectory = Convert-Path .
-$installAgentScriptLocation = Join-Path -Path $currentDirectory -ChildPath "TestAgentInstall.ps1"
-Write-Verbose "installAgentScriptLocation = $installAgentScriptLocation"
-
-$configureTestAgentScriptLocation = Join-Path -Path $currentDirectory -ChildPath "TestAgentConfiguration.ps1"
-Write-Verbose "configureTestAgentScriptLocation = $configureTestAgentScriptLocation"
-
-$checkAgentInstallationScriptLocation = Join-Path -Path $currentDirectory -ChildPath "CheckTestAgentInstallation.ps1"
-Write-Verbose "checkAgentInstallationScriptLocation = $checkAgentInstallationScriptLocation"
-
-$downloadTestAgentScriptLocation = Join-Path -Path $currentDirectory -ChildPath "DownloadTestAgent.ps1"
-Write-Verbose "downloadTestAgentScriptLocation = $downloadTestAgentScriptLocation"
-
-$verifyTestMachinesAreInUse = Join-Path -Path $currentDirectory -ChildPath "VerifyTestMachinesAreInUse.ps1"
-Write-Verbose "VerifyTestMachinesAreInUseScriptLocation = $verifyTestMachinesAreInUse"
-
-# Import the Task.Internal dll that has all the cmdlets we need for Build
-import-module "Microsoft.TeamFoundation.DistributedTask.Task.Internal"
-import-module "Microsoft.TeamFoundation.DistributedTask.Task.Common"
-import-module "Microsoft.TeamFoundation.DistributedTask.Task.DTA"
-import-module "Microsoft.TeamFoundation.DistributedTask.Task.DevTestLabs"
-
-Write-Verbose "Getting the connection object"
-$connection = Get-VssConnection -TaskContext $distributedTaskContext
-
-Write-Verbose "Getting Personal Access Token for the Run"
-$vssEndPoint = Get-ServiceEndPoint -Context $distributedTaskContext -Name "SystemVssConnection"
-$personalAccessToken = Get-PersonalAccessToken $vssEndpoint
+[CmdletBinding()]
+param()
+ 
+Trace-VstsEnteringInvocation $MyInvocation
+try {
+    Import-VstsLocStrings "$PSScriptRoot\Task.json"
+
+    $testMachineGroup           = Get-VstsInput -Name testMachineGroup -Require
```

Comment:
>$testMachineGroup           = Get-VstsInput -Name testMachineGroup -Require [](start = 4, length = 75)

weren't we thinking of adding an option instead of this being a free for all text box ?